onnxscript/values.py

```diff
@@ -183,77 +183,73 @@ def fun(*args, **kwargs):
 
         return fun
 
+    def wrap(self, inputs):
+        """Adapts inputs into representation used by onnxscript eager mode.
+
+        This does the following transformations:
+        * It adds an onnxscript Tensor wrapper around numpy arrays, which
+        allows the use of overloaded operators like + to be controlled by onnxscript.
+        * It also provides a promotion of scalars into tensors as a convenience.
+        This is needed to complement the similar promotion supported by the
+        onnxscript converter (for example, when an attribute is promoted and used
+        as an input argument).
+
+        Args:
+            inputs: a list/tuple of inputs to an ONNX function
+
+        Returns:
+            a pair (wrapped_inputs, flag) where flag indicates whether any numpy array
+            was wrapped into a Tensor.
+        """
+        has_array = False
+
+        def do_wrap(input):
```

Comment:
Looks like do_wrap do not need to clse over local variables? If that's the case I'd prefer module level functions which are easier to test and maintain

Reply:
Yes, I considered that. It might be better to do that. (I ended up with this choice in case we want to define derived classes where we want to alter this behavior by overriding this method. But I don't see any immediate use for such overriding.)